doc/mid-based-signalling.md

```diff
@@ -0,0 +1,252 @@
+# MID Based Signalling
+
+We intend to migrate away from refering to tracks by WebRTC track ID since this has known
+problems with ID consistency on either side of the WebRTC connection (we currently parse the
+SDP to work around this).
+
+We also propose switch the structure of MSC3401 call events to be compatible with extensible
+events. Legacy calls would have to remain the same for backwards compatibility. It is probably
+impractical to have any call events support both formats because this would mean the SDP, which
+can be quite large,would have to be duplicated in the JSON. Therefore, there's a significant
+advantage to taking to opportunity to migrate to an extensible event format now.
+
+We use the same event types for call invites and answers, so it is important that these events
+are only send via to device message directly to client known to support them. They must not be
+sent in rooms not supporting extensible events.
+
+This outlines how a switch to mid + media UUID based signalling would work in more detail.
+
+**Note: I've made this mixin about describing the tracks that are being sent on the transceivers
+and the keys are the mids rather than the media UUIDs. Hoping that sketching this out will give
+us an idea of whether it could work or not.**
+
+m.call.invite:
+```
+{
+    "m.negotiate": {
+        "version": 1,
+        "call_id": "35657a5b793ce",
+        "conf_id": "bbe53499f82e3",
+        "invitee": "@bob:example.org",
+        "lifetime": 60000,
+        "party_id": "123456",
+        "description": {
+            "type": "offer",
+            "sdp": "[...]",
+        }
+    },
```

Comment:
Perhaps this should also be split up into several pieces?

Reply:
As in split the common call stuff out from the actual negotiation? Interesting idea.

Reply:
Well `m.call.candidates` would still include some of this, so it would make sense to have that in a common mixin